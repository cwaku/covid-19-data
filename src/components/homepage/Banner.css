.banner {
  background-color: #fafafa;
  padding: 0;
  margin: 0;
  height: 500px;
  background-image: url(../../images/banner.jpg);
  background-repeat: no-repeat;
  background-size: contain;
  background-attachment: fixed;
  margin-bottom: 40px;
  color: #000;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.cases {
  font-size: 28px;
}


h1,
h2 {
  margin: 0;
}
