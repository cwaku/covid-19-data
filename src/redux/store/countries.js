import moment from 'moment';
import { createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
// set axios default url
axios.defaults.baseURL = 'https://api.covid19tracking.narrativa.com/api/';

const date = moment(new Date()).format('YYYY-MM-DD');

const countriesSlice = createSlice({
  name: 'countries',
  initialState: {
    countries: [],
    selectedCountry: '',
    searchCountry: '',
  },
  reducers: {
    getCountriesSuccess: (state, action) => ({ ...state, countries: action.payload }),
    getSelectedCountrySuccess: (state, action) => ({ ...state, selectedCountry: action.payload }),
    getSearchedCountrySuccess: (state, action) => ({ ...state, searchCountry: action.payload }),
    getCountriesFailure: (state, action) => ({ ...state, countries: action.payload }),
  },
});

export default countriesSlice.reducer;

// Actions

export const {
  getCountriesSuccess,
  getSelectedCountrySuccess,
  getSearchedCountrySuccess,
  getCountriesFailure,
} = countriesSlice.actions;

export const fetchCountries = () => async (dispatch) => {
  try {
    const response = await axios.get(date);
    if (response) {
      const { dates, total } = response.data;
      const { countries } = dates[date];
      const countriesList = Object.values(countries);

      const singleCountry = countriesList.map((country) => (
        {
          name: country.name,
          confirmedCases: country.today_confirmed,
          deaths: country.today_deaths,
          recovered: country.today_recovered,
          active: country.today_open_cases,
          id: country.id,
        }
      ));

      const totalConfirmedCases = total.today_confirmed;
      const allData = { singleCountry, totalConfirmedCases };
      dispatch(getCountriesSuccess(allData));
    }
  } catch (error) {
    dispatch(getCountriesFailure(error));
  }
};
