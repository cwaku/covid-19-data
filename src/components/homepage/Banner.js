import * as React from 'react';
import Box from '@mui/material/Box';

import './Banner.css';
import { useSelector } from 'react-redux';

const Banner = () => {
  const total = useSelector(state => state.countries.countries.totalConfirmedCases);

  return (
    <Box sx={{ width: '100%' }}
      className="banner">
      <h1>COVID-19</h1>
      <h2>Coronavirus disease (COVID-19)</h2>
      <h2 className="cases">Total confirmed cases: {total}</h2>
      <h3 className="no-margin">Coronavirus disease (COVID-19) is an infectious disease caused by a newly discovered coronavirus.</h3>
      <h3 className="no-margin">Most people infected with the COVID-19 virus will experience mild to moderate respiratory illness and recover without requiring special treatment.</h3>
      <h3 className="no-margin">The COVID-19 virus spreads primarily through droplets of saliva or discharge from the nose when an infected person coughs or sneezes, so itâ€™s important that you also practice respiratory etiquette (for example, by coughing into a flexed elbow).</h3>
      <h3 className="no-margin">At this time, there are no specific vaccines or treatments for COVID-19.</h3>
      <h3 className="no-margin">However, there are many ongoing clinical trials evaluating potential treatments. WHO will continue to provide updated information as soon as clinical findings become available.</h3>
      <h3 className="no-margin">The World Health Organization (WHO) is the authority on the COVID-19 pandemic.</h3>
      <h3 className="no-margin">Read more about the COVID-19 pandemic at <a href="https://www.who.int/emergencies/diseases/novel-coronavirus-2019">https://www.who.int/emergencies/diseases/novel-coronavirus-2019</a></h3>
    </Box>
  );
}

export default Banner;