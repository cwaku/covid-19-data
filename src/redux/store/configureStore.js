import { configureStore } from '@reduxjs/toolkit'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import { combineReducers } from 'redux'
import countries from './countries'

const reducer = combineReducers({
  // reducers go here
  countries,
})

const store = configureStore({
  reducer,
  middleware: [thunk, logger],
})

export default store;