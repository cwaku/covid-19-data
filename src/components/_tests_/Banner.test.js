import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../../redux/store/configureStore';
import Banner from '../homepage/Banner';

describe('Header component', () => {
  it('should render the  banner correctly ', () => {
    const header = render(
      <BrowserRouter>
        <Provider store={store}>
          <Banner />
        </Provider>
      </BrowserRouter>,
    );

    expect(header).toMatchSnapshot();
  });
});
