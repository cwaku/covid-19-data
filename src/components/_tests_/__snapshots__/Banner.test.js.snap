// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render the  banner correctly  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="banner MuiBox-root css-8atqhb"
      >
        <h1>
          COVID-19
        </h1>
        <h2>
          Coronavirus disease (COVID-19)
        </h2>
        <h2
          class="cases"
        >
          Total confirmed cases:
        </h2>
        <div
          class="banner-text"
        >
          <h3
            class="no-margin"
          >
            Coronavirus disease (COVID-19) is an infectious disease caused by a newly discovered coronavirus.
          </h3>
          <h3
            class="no-margin"
          >
            Most people infected with the COVID-19 virus will experience mild to moderate respiratory illness and recover without requiring special treatment.
          </h3>
          <h3
            class="no-margin"
          >
            The COVID-19 virus spreads primarily through droplets of saliva or discharge from the nose when an infected person coughs or sneezes, so it’s important that you also practice respiratory etiquette (for example, by coughing into a flexed elbow).
          </h3>
          <h3
            class="no-margin"
          >
            At this time, there are no specific vaccines or treatments for COVID-19.
          </h3>
          <h3
            class="no-margin"
          >
            However, there are many ongoing clinical trials evaluating potential treatments. WHO will continue to provide updated information as soon as clinical findings become available.
          </h3>
          <h3
            class="no-margin"
          >
            The World Health Organization (WHO) is the authority on the COVID-19 pandemic.
          </h3>
          <h3
            class="no-margin"
          >
            Read more about the COVID-19 pandemic at
            <a
              href="https://www.who.int/emergencies/diseases/novel-coronavirus-2019"
            >
              https://www.who.int/emergencies/diseases/novel-coronavirus-2019
            </a>
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="banner MuiBox-root css-8atqhb"
    >
      <h1>
        COVID-19
      </h1>
      <h2>
        Coronavirus disease (COVID-19)
      </h2>
      <h2
        class="cases"
      >
        Total confirmed cases:
      </h2>
      <div
        class="banner-text"
      >
        <h3
          class="no-margin"
        >
          Coronavirus disease (COVID-19) is an infectious disease caused by a newly discovered coronavirus.
        </h3>
        <h3
          class="no-margin"
        >
          Most people infected with the COVID-19 virus will experience mild to moderate respiratory illness and recover without requiring special treatment.
        </h3>
        <h3
          class="no-margin"
        >
          The COVID-19 virus spreads primarily through droplets of saliva or discharge from the nose when an infected person coughs or sneezes, so it’s important that you also practice respiratory etiquette (for example, by coughing into a flexed elbow).
        </h3>
        <h3
          class="no-margin"
        >
          At this time, there are no specific vaccines or treatments for COVID-19.
        </h3>
        <h3
          class="no-margin"
        >
          However, there are many ongoing clinical trials evaluating potential treatments. WHO will continue to provide updated information as soon as clinical findings become available.
        </h3>
        <h3
          class="no-margin"
        >
          The World Health Organization (WHO) is the authority on the COVID-19 pandemic.
        </h3>
        <h3
          class="no-margin"
        >
          Read more about the COVID-19 pandemic at
          <a
            href="https://www.who.int/emergencies/diseases/novel-coronavirus-2019"
          >
            https://www.who.int/emergencies/diseases/novel-coronavirus-2019
          </a>
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
